src/Admin/AmpPlugins.php

```diff
@@ -0,0 +1,332 @@
+			'donate_link'              => '',
+			'icons'                    => [
+				'1x'  => '',
+				'2x'  => '',
+				'svg' => '',
+			],
+			'wporg'                    => false,
+		];
+
+		$plugin['ratings'] = ( ! empty( $plugin['ratings'] ) && is_array( $plugin['ratings'] ) ) ? $plugin['ratings'] : [];
+		$plugin['ratings'] = $plugin['ratings'] + $default['ratings'];
+
+		$plugin['icons'] = ( ! empty( $plugin['icons'] ) && is_array( $plugin['icons'] ) ) ? $plugin['icons'] : [];
+		$plugin['icons'] = wp_parse_args( $plugin['icons'], $default['icons'] );
+
+		return wp_parse_args( $plugin, $default );
+	}
+
+	/**
+	 * Adds hooks.
+	 *
+	 * @return void
+	 */
+	public function register() {
+
+		$screen = get_current_screen();
+		if ( $screen instanceof WP_Screen && 'plugin-install' === $screen->id ) {
+			add_action( 'admin_enqueue_scripts', [ $this, 'enqueue_scripts' ] );
+		}
+
+		add_filter( 'install_plugins_tabs', [ $this, 'add_tab' ] );
+		add_filter( 'install_plugins_table_api_args_' . self::AMP_COMPATIBLE, [ $this, 'filter_plugins_table_api_args' ] );
+		add_filter( 'plugins_api', [ $this, 'filter_plugins_api' ], 10, 3 );
+		add_filter( 'plugin_install_action_links', [ $this, 'filter_action_links' ], 10, 2 );
+		add_filter( 'plugin_row_meta', [ $this, 'filter_plugin_row_meta' ], 10, 3 );
+
+		add_action( 'install_plugins_' . self::AMP_COMPATIBLE, 'display_plugins_table' );
+	}
+
+	/**
+	 * Enqueue style for plugin install page.
+	 *
+	 * @return void
+	 */
+	public function enqueue_scripts() {
+
+		$asset_file   = AMP__DIR__ . '/assets/js/' . self::ASSET_HANDLE . '.asset.php';
+		$asset        = require $asset_file;
+		$dependencies = $asset['dependencies'];
+		$version      = $asset['version'];
+
+		wp_enqueue_script(
+			self::ASSET_HANDLE,
+			amp_get_asset_url( 'js/' . self::ASSET_HANDLE . '.js' ),
+			$dependencies,
+			$version,
+			true
+		);
+
+		wp_enqueue_style(
+			'amp-admin',
+			amp_get_asset_url( 'css/amp-admin.css' ),
+			[],
+			AMP__VERSION
+		);
+
+		$js_data = [
+			'AMP_COMPATIBLE' => self::AMP_COMPATIBLE,
+			'AMP_PLUGINS'    => wp_list_pluck( $this->get_plugins(), 'slug' ),
+		];
+
+		wp_add_inline_script(
+			self::ASSET_HANDLE,
+			sprintf(
+				'var ampPlugins = %s;',
+				wp_json_encode( $js_data )
+			),
+			'before'
+		);
+	}
+
+	/**
+	 * Add extra tab in plugin install screen.
+	 *
+	 * @param array $tabs List of tab in plugin install screen.
+	 *
+	 * @return array List of tab in plugin install screen.
+	 */
+	public function add_tab( $tabs ) {
+
+		return array_merge(
+			$tabs,
+			[
+				self::AMP_COMPATIBLE => esc_html__( 'AMP Compatible', 'amp' ),
+			]
+		);
+	}
+
+	/**
+	 * Modify args for the plugins_api query on the AMP-compatible tab in plugin install screen.
+	 *
+	 * @return array
+	 */
+	public function filter_plugins_table_api_args() {
+
+		$per_page   = 100; // @todo There are currently 56 plugins, so this will show all. This is done because pagination is not working.
```

Comment:
@dhaval-parekh Please look into pagination of the AMP-compatible plugins list in another PR.

Reply:
Please have a look at this PR #6681 
I have addressed the pagination issue in this PR.